crates/sui-bridge-indexer-alt/src/handlers/token_transfer_handler.rs

```diff
@@ -0,0 +1,282 @@
+                            results.push(TokenTransferDataWrapper::from_approve_event(
+                                bcs::from_bytes(&ev.contents)?,
+                                tx,
+                                block_height,
+                                timestamp_ms,
+                            ));
+                        } else if self.claimed_event_type == ev.type_ {
+                            info!("Observed Sui Claim {:?}", ev);
+                            // todo: metrics.total_sui_token_transfer_claimed.inc();
+                            results.push(TokenTransferDataWrapper::from_claimed_event(
+                                bcs::from_bytes(&ev.contents)?,
+                                tx,
+                                block_height,
+                                timestamp_ms,
+                            ));
+                        }
+
+                        Ok(results)
+                    },
+                )
+            })
+    }
+}
+
+#[async_trait]
+impl Handler for TokenTransferHandler {
+    type Store = Db;
+
+    async fn commit<'a>(
+        values: &[Self::Value],
+        conn: &mut <Self::Store as Store>::Connection<'a>,
+    ) -> sui_indexer_alt_framework::Result<usize> {
+        let (token_transfers, data) = values.iter().fold(
+            (vec![], vec![]),
+            |(mut token_transfers, mut data_vec),
+             TokenTransferDataWrapper {
+                 token_transfer,
+                 data,
+             }| {
+                token_transfers.push(token_transfer.clone());
+                if let Some(d) = data {
+                    data_vec.push(d.clone());
+                }
+                (token_transfers, data_vec)
+            },
+        );
+
+        conn.transaction(|conn| {
+            async move {
+                {
+                    use token_transfer::columns::*;
+                    diesel::insert_into(token_transfer::table)
+                        .values(&token_transfers)
+                        .on_conflict((chain_id, nonce, status))
+                        .do_update()
+                        .set((
+                            chain_id.eq(excluded(chain_id)),
+                            nonce.eq(excluded(nonce)),
+                            status.eq(excluded(status)),
+                            block_height.eq(excluded(block_height)),
+                            timestamp_ms.eq(excluded(timestamp_ms)),
+                            txn_hash.eq(excluded(txn_hash)),
+                            txn_sender.eq(excluded(txn_sender)),
+                            gas_usage.eq(excluded(gas_usage)),
+                            data_source.eq(excluded(data_source)),
+                            is_finalized.eq(excluded(is_finalized)),
+                        ))
+                        .filter(is_finalized.eq(false))
+                        .execute(conn)
+                        .await?;
+                }
+
+                {
```

Comment:
Typically we would try to split this out into its own pipeline rather than coupling the writes. Is that not possible here?

Reply:
I will split them into 2 pipelines